include/envoy/network/transport_socket.h

```diff
@@ -136,6 +138,30 @@ class TransportSocket {
 
 typedef std::unique_ptr<TransportSocket> TransportSocketPtr;
 
+/**
+ * Options for creating transport sockets.
+ */
+class TransportSocketOptions {
+public:
+  virtual ~TransportSocketOptions() {}
+
+  /**
+   * @return the const optional server name to set in the transport socket, for example SNI for
+   *         SSL, regardless of the upstream cluster configuration. The filters like tcp_proxy
+   *         should override the server name in the upstream cluster configuration with that value.
+   */
+  virtual const absl::optional<std::string>& overrideServerName() const PURE;
+
+  /**
+   * @param vector of bytes to which the option should append hash key data that will be used
+   *        to separate connections based on the option. Any data already in the key vector must
+   *        not be modified.
+   */
+  virtual void hashKey(std::vector<uint8_t>& key) const PURE;
```

Comment:
Rename to `appendHashKey` to indicate this is mutating.

Reply:
@htuch This is done to be similar to Socket::Option::hashKey() https://github.com/envoyproxy/envoy/blob/15706964a29e595c045a5d7ef0646d04f347dcc1/include/envoy/network/listen_socket.h#L53

Do you understand what the function should do according to the documentation of `Socket::Option::hashKey()`?

>   @param vector of bytes to which the option should append hash key data that will be used
      to separate connections based on the option. Any data already in the key vector must
             not be modified.

Reply:
Yeah, fair enough, for consistency it's reasonable to keep this convention then.